test/groovy/com/sap/piper/integration/TransportManagementServiceTest.groovy

```diff
@@ -62,12 +62,69 @@ class TransportManagementServiceTest extends BasePiperTest {
         def clientSecret = 'mySecret'
 
         def tms = new TransportManagementService(nullScript, [verbose: true])
-        tms.authentication(uaaUrl, clientId, clientSecret)
+        def token = tms.authentication(uaaUrl, clientId, clientSecret)
 
         assertThat(loggingRule.log, containsString("[TransportManagementService] OAuth Token retrieval started."))
         assertThat(loggingRule.log, containsString("[TransportManagementService] UAA-URL: '${uaaUrl}', ClientId: '${clientId}'"))
         assertThat(loggingRule.log, containsString("Received response '{ \"access_token\": \"myOAuthToken\" }' with status 200."))
         assertThat(loggingRule.log, containsString("[TransportManagementService] OAuth Token retrieved successfully."))
+        assertThat(token, is('myOAuthToken'))
+    }
+
+    @Test
+    void retrieveOAuthToken__failure() {
+        Map requestParams
+        helper.registerAllowedMethod('httpRequest', [Map.class], { m ->
+            requestParams = m
+            return [content: 'Here an error message is expected', status: 400]
+        })
+
+        def uaaUrl = 'http://dummy.com/oauth'
+        def clientId = 'myId'
+        def clientSecret = 'mySecret'
+
+        def tms = new TransportManagementService(nullScript, [:])
+        def errorCaught = false
+        try {
+            tms.authentication(uaaUrl, clientId, clientSecret)
+        } catch (e) {
+            errorCaught = true
+            assertThat(e.getMessage(), is("OAuth Token retrieval failed. Consider re-running in verbose mode in order to get more details."))
+        } finally {
+            assertThat(errorCaught, is(true))
+            assertThat(loggingRule.log, containsString("[TransportManagementService] OAuth Token retrieval started."))
+            assertThat(requestParams, hasEntry('url', "${uaaUrl}/oauth/token/?grant_type=client_credentials&response_type=token"))
+            assertThat(requestParams, hasEntry('requestBody', "grant_type=password&username=${clientId}&password=${clientSecret}".toString()))
+            assertThat(requestParams.customHeaders[1].value, is("Basic ${"${clientId}:${clientSecret}".bytes.encodeBase64()}"))
+        }
+    }
+
+    @Test
+    void retrieveOAuthToken__failure__status__400__inVerboseMode() {
+        Map requestParams
+        helper.registerAllowedMethod('httpRequest', [Map.class], { m ->
+            requestParams = m
+            return [content: 'Here an error message is expected', status: 400]
+        })
+        
+        def uaaUrl = 'http://dummy.com/oauth'
+        def clientId = 'myId'
+        def clientSecret = 'mySecret'
+        
+        def tms = new TransportManagementService(nullScript, [verbose: true])
+        def errorCaught = false
+        try {
+            tms.authentication(uaaUrl, clientId, clientSecret)
+        } catch (e) {
```

Comment:
See above. With `ExpectedException` and `loggingRule.expect(./.)` there is no need for such a way of asserting the details in case we get an exception.